osutil/epoll/epoll.go

```diff
@@ -0,0 +1,157 @@
+// Package epoll contains a thin wrapper around the epoll(7) facility.
+//
+// Using epoll from Go is unusual as the language provides facilities that
+// normally make using it directly pointless. Epoll is strictly required for
+// unusual kernel interfaces that use event notification but don't implement
+// file descriptors that provide usual read/write semantics.
+//
+// It might be possible to remove this code and use internal/poll and
+// syscall.RawConn but the necessary interfaces require golang 1.12 that snapd
```

Comment:
@mvo5 is this something I should look into, now that we're on a newer golang version?

Reply:
would be worth taking a look but not sure what is meant here as we can't access an internal interface